qiskit_ibm/backend_group.py

```diff
@@ -0,0 +1,134 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A hub, group and project in an IBM Quantum account."""
+
+import logging
+from collections import OrderedDict
+import traceback
+from typing import Any, Dict, Optional
+
+from qiskit.providers.backend import BackendV1 as Backend
+from qiskit.providers.models import PulseBackendConfiguration, QasmBackendConfiguration
+from qiskit_ibm import ibm_provider  # pylint: disable=unused-import
+from qiskit_ibm.exceptions import IBMInputValueError
+
+from .utils.json_decoder import decode_backend_configuration
+from .ibm_backend import IBMBackend, IBMSimulator
+from .api.clients import AccountClient
+from .credentials import Credentials
+
+logger = logging.getLogger(__name__)
+
+
+class BackendGroup:
+    """Represents a group of IBM Quantum backends that belong to a single hub/group/project."""
```

Comment:
So I suggested `BackendGroup` because the old code only dealt with backends. This is no longer the case, as it now also deals with services. This now represents more about what a hub/group/project entitles you to. So unfortunately I think `BackendGroup` is now a bad name for it. Can this be merged with the existing `HubGroupProject` class? Or maybe rename it to something like `Entitlement`?

Reply:
I have renamed existing `HubGroupProject` class to `HubGroupProjectID` and `BackendGroup` class to `HubGroupProject`. Let me know your thoughts.